js/components/operatorSignIn/operatorSelection.tsx

```diff
@@ -1,16 +1,40 @@
+import { ApiResult } from "../../api";
+import { findEmployeeByBadgeSerial } from "../../hooks/useEmployees";
+import { useNfc } from "../../hooks/useNfc";
+import { Employee } from "../../models/employee";
 import { className } from "../../util/dom";
-import { ReactElement, useId, useState } from "react";
+import { ReactElement, useEffect, useId, useState } from "react";
 
 export const OperatorSelection = ({
   nfcSupported: nfcSupported,
   onOK,
+  employees,
 }: {
   nfcSupported: boolean;
   onOK: (badge: string) => void;
+  employees: ApiResult<Employee[]>;
 }): ReactElement => {
   const inputId = useId();
   const [badgeEntry, setBadgeEntry] = useState<string>("");
 
+  const nfcAbortController = new AbortController();
+  const nfcResult = useNfc(nfcAbortController);
+
+  useEffect(() => {
+    if (nfcResult.status === "success" && employees.status === "ok") {
+      const badge = findEmployeeByBadgeSerial(
+        employees.result,
+        nfcResult.data,
+      )?.badge;
+
+      if (badge) {
+        onOK(badge);
+      } else {
+        // handle case where we can't find operator for badge
```

Comment:
Is this going to be in a subsequent PR?

Reply:
Yes, my thought process is that this would be best implemented once your work is ready, since you're also making additions to handle the state of the sign in flow.